packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -350,6 +360,19 @@ class VisualStudio {
     visualStudioDetails ??= _visualStudioDetails(
         validateRequirements: true,
         additionalArguments: <String>[...minimumVersionArguments, _vswherePrereleaseArgument]);
+    // If Visual Studio IDE is not found, try searching for Build Tools.
```

Comment:
I think the better order would be: stable VS, stable BT, pre-release VS, pre-release BT.